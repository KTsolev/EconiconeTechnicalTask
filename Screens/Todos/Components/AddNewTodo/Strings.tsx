export const inputPlaceHolder = 'Enter new task';
export const editTask = 'Edit Task';
export const addTask = 'Add Task';
export const taskError = 'Name of task is mandatory!';
