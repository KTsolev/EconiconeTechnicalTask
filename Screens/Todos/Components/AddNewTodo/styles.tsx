import {StyleSheet} from 'react-native';

export const styles = StyleSheet.create({
  container: {
    flex: 1,
    paddingHorizontal: 10,
    marginVertical: 4,
    paddingVertical: 14,
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'center',
  },
  inputHolder: {
    flex: 0.82,
    position: 'relative',
    alignItems: 'center',
    justifyContent: 'center',
    borderRadius: 6,
    marginHorizontal: 4,
    borderWidth: 1,
    borderColor: '#888888',
  },
  text: {
    flex: 0.8,
    fontSize: 16,
    paddingLeft: 4,
    color: '#282C34',
  },
  input: {
    maxHeight: 120,
    minHeight: 40,
    color: '#888888',
    fontWeight: '600',
    padding: 10,
  },
  errorText: {
    position: 'absolute',
    top: 44,
    left: 45,
    color: '#ff2200',
    fontSize: 12,
  },
  inputError: {
    borderColor: '#ff2200',
  },
  button: {
    flex: 0.5,
    maxWidth: 184,
    maxHeight: 68,
    minHeight: 40,
    padding: 10,
    marginHorizontal: 4,
    borderRadius: 6,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#1c90ff',
  },
  inActiveButton: {
    backgroundColor: '#c8c8c8',
  },
  buttonText: {
    textAlign: 'center',
    fontWeight: '600',
    color: '#fff',
  },
});
