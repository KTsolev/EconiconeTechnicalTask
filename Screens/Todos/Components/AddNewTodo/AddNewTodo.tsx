import React, {FunctionComponent, useEffect, useState} from 'react';
import {View, TextInput, TouchableOpacity, Text} from 'react-native';
import {TodoModel} from '../../Models/TodoModel';
import {useDispatch} from 'react-redux';
import {addTodo, editTodo} from '../../Todoslice/TodoSlice';
import CheckBox from '@react-native-community/checkbox';
import uniqid from 'uniqid';
import {styles} from './styles';
import * as Strings from './Strings';

export const AddNewItem: FunctionComponent<AddNewItemProps> = ({
  item,
  closeEditTask,
}): JSX.Element => {
  const [isCompleted, setIsCompleted] = useState(false);
  const [isEdit, setIsEdit] = useState(false);
  const [hasError, setHasErrors] = useState(false);
  const [todoName, setTodoName] = useState('');
  const [todoError, setTodoError] = useState('');
  const dispatch = useDispatch();

  useEffect(() => {
    if (item !== null && item !== undefined) {
      setIsEdit(true);
      setTodoName(item.title);
    } else {
      setIsEdit(false);
      setIsCompleted(false);
      setTodoName('');
    }
  }, [item]);

  const createNewOne = (): void => {
    if (todoName !== '') {
      setHasErrors(false);
      setTodoError('');
      if (isEdit) {
        dispatch(
          editTodo(new TodoModel(item!.id, todoName, isCompleted, true)),
        );

        closeEditTask && closeEditTask();
      } else {
        dispatch(addTodo(new TodoModel(uniqid(), todoName, isCompleted, true)));
      }
    } else {
      setHasErrors(true);
      setTodoError(Strings.taskError);
    }
  };

  const checkIfErrors = (): void => {
    if (todoName !== '') {
      setHasErrors(false);
      setTodoError('');
    } else {
      setHasErrors(true);
      setTodoError(Strings.taskError);
    }
  };
  return (
    <View key={item?.id || 111} style={styles.container}>
      {isEdit ? (
        <CheckBox value={isCompleted} onValueChange={setIsCompleted} />
      ) : null}
      <View style={styles.inputHolder}>
        <TextInput
          style={[styles.input]}
          value={todoName}
          placeholder={Strings.inputPlaceHolder}
          onChangeText={setTodoName}
          onEndEditing={checkIfErrors}
          onBlur={checkIfErrors}
          keyboardType="default"
        />

        {hasError && <Text style={styles.errorText}>{todoError}</Text>}
      </View>
      <TouchableOpacity
        style={[styles.button, hasError ? styles.inActiveButton : null]}
        onPress={createNewOne}
        disabled={hasError}>
        <Text style={styles.buttonText}>
          {isEdit ? Strings.editTask : Strings.addTask}
        </Text>
      </TouchableOpacity>
    </View>
  );
};

type AddNewItemProps = {
  item?: TodoModel | null;
  closeEditTask?: () => void;
};
