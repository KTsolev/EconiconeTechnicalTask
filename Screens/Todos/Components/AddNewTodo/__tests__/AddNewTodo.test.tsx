import React from 'react';
import {AddNewItem} from '../AddNewTodo';
import renderer from 'react-test-renderer';
import {fireEvent, render} from '@testing-library/react-native';
import {TodoModel} from '../../../Models/TodoModel';
import {todos} from '../../../../../__mocks__/todos.mock';

const mockDispatchFn = jest.fn();
const mockCloseEdit = jest.fn();

jest.mock('uniqid', () => () => '1911caln7e0l28wbt2j');

jest.mock('react-redux', () => ({
  useSelector: (selector: () => any) => selector(),
  useDispatch: () => mockDispatchFn,
}));

describe('AddNewItem tests', () => {
  beforeEach(() => {
    jest.resetAllMocks();
  });

  afterEach(() => {
    jest.resetAllMocks();
  });

  it('should render component', async () => {
    const tree = renderer.create(<AddNewItem />);
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('should handle user input', async () => {
    const {getByTestId, toJSON} = render(<AddNewItem />);
    const input = getByTestId('input');
    fireEvent.changeText(input, 'sampleTask4');
    expect(toJSON()).toMatchSnapshot();
  });

  it('should handle error after incorrect user input', async () => {
    const {getByTestId, toJSON} = render(<AddNewItem />);
    const input = getByTestId('input');
    const button = getByTestId('createButton');

    fireEvent.changeText(input, '');
    fireEvent.press(button);
    expect(toJSON()).toMatchSnapshot();
    expect(mockDispatchFn).not.toHaveBeenCalled();
  });

  it('should handle check for errors after incorrect user input', async () => {
    const {getByTestId, toJSON} = render(<AddNewItem />);
    const input = getByTestId('input');
    fireEvent.changeText(input, '');
    fireEvent(input, 'onBlur', {nativeEvent: {}});

    expect(toJSON()).toMatchSnapshot();
    expect(mockDispatchFn).not.toHaveBeenCalled();
  });

  it('should handle check for errors after incorrect user input and then correct is given', async () => {
    const {getByTestId, toJSON} = render(<AddNewItem />);
    const input = getByTestId('input');
    fireEvent.changeText(input, '');
    fireEvent(input, 'onBlur', {nativeEvent: {}});
    fireEvent.changeText(input, 'samplee task 2');
    fireEvent(input, 'onBlur', {nativeEvent: {}});

    expect(toJSON()).toMatchSnapshot();
    expect(mockDispatchFn).not.toHaveBeenCalled();
  });

  it('should remove error due to incorrect user input, after correct one is given', async () => {
    const {getByTestId, toJSON} = render(<AddNewItem />);
    const input = getByTestId('input');
    const button = getByTestId('createButton');

    fireEvent.changeText(input, '');
    fireEvent.press(button);
    fireEvent.changeText(input, 'sampletask22');

    expect(toJSON()).toMatchSnapshot();
    expect(mockDispatchFn).not.toHaveBeenCalled();
  });

  it('should handle add new after correct user input', async () => {
    const {getByTestId, toJSON} = render(<AddNewItem />);
    const input = getByTestId('input');
    const button = getByTestId('createButton');

    fireEvent.changeText(input, 'sample task 1');
    fireEvent.press(button);
    expect(toJSON()).toMatchSnapshot();
    expect(mockDispatchFn).toHaveBeenCalledWith({
      payload: new TodoModel(
        '1911caln7e0l28wbt2j',
        'sample task 1',
        false,
        true,
      ),
      type: 'todos/addTodo',
    });
  });
});

describe('AddNewItem edit mode tests', () => {
  beforeEach(() => {
    jest.resetAllMocks();
  });

  afterEach(() => {
    jest.resetAllMocks();
  });

  it('should render component', async () => {
    const tree = renderer.create(
      <AddNewItem item={todos[2]} closeEditTask={mockCloseEdit} />,
    );
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('should handle user input', async () => {
    const {getByTestId, toJSON} = render(
      <AddNewItem item={todos[2]} closeEditTask={mockCloseEdit} />,
    );
    const input = getByTestId('input');
    fireEvent.changeText(input, 'sampleTask4');
    expect(toJSON()).toMatchSnapshot();
  });

  it('should handle error after incorrect user input', async () => {
    const {getByTestId, toJSON} = render(
      <AddNewItem item={todos[2]} closeEditTask={mockCloseEdit} />,
    );
    const input = getByTestId('input');
    const button = getByTestId('createButton');

    fireEvent.changeText(input, '');
    fireEvent.press(button);
    expect(toJSON()).toMatchSnapshot();
    expect(mockDispatchFn).not.toHaveBeenCalled();
    expect(mockCloseEdit).not.toHaveBeenCalled();
  });

  it('should handle add new after correct user input', async () => {
    const {getByTestId, toJSON} = render(
      <AddNewItem item={todos[2]} closeEditTask={mockCloseEdit} />,
    );
    const input = getByTestId('input');
    const button = getByTestId('createButton');

    fireEvent.changeText(input, 'sample task 1');
    fireEvent.press(button);
    expect(toJSON()).toMatchSnapshot();
    expect(mockCloseEdit).toHaveBeenCalled();
    expect(mockDispatchFn).toHaveBeenCalledWith({
      payload: new TodoModel('l28uy38o', 'sample task 1', false, true),
      type: 'todos/editTodo',
    });
  });
});
