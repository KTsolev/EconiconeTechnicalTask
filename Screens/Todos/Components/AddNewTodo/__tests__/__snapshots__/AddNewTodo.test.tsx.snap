// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddNewItem edit mode tests should handle add new after correct user input 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "row",
      "justifyContent": "center",
      "marginVertical": 4,
      "paddingHorizontal": 10,
      "paddingVertical": 14,
    }
  }
>
  <View
    accessibilityRole="checkbox"
    accessibilityState={
      Object {
        "checked": false,
        "disabled": false,
      }
    }
    accessibilityValue={
      Object {
        "text": "off",
      }
    }
    accessible={true}
    pointerEvents="auto"
    testID="completeEdit"
  >
    <RNCCheckbox
      forwardedRef={null}
      onValueChange={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "height": 32,
            "width": 32,
          },
          undefined,
        ]
      }
      value={false}
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderColor": "#888888",
        "borderRadius": 6,
        "borderWidth": 1,
        "flex": 0.82,
        "justifyContent": "center",
        "marginHorizontal": 4,
        "position": "relative",
      }
    }
  >
    <TextInput
      keyboardType="default"
      onBlur={[Function]}
      onChangeText={[Function]}
      onEndEditing={[Function]}
      placeholder="Enter new task"
      style={
        Array [
          Object {
            "color": "#888888",
            "fontWeight": "600",
            "maxHeight": 120,
            "minHeight": 40,
            "padding": 10,
          },
        ]
      }
      testID="input"
      value="sample task 1"
    />
  </View>
  <View
    accessibilityState={
      Object {
        "disabled": false,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#1c90ff",
        "borderRadius": 6,
        "flex": 0.5,
        "justifyContent": "center",
        "marginHorizontal": 4,
        "maxHeight": 68,
        "maxWidth": 184,
        "minHeight": 40,
        "opacity": 1,
        "padding": 10,
      }
    }
    testID="createButton"
  >
    <Text
      style={
        Object {
          "color": "#fff",
          "fontWeight": "600",
          "textAlign": "center",
        }
      }
    >
      Edit Task
    </Text>
  </View>
</View>
`;

exports[`AddNewItem edit mode tests should handle error after incorrect user input 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "row",
      "justifyContent": "center",
      "marginVertical": 4,
      "paddingHorizontal": 10,
      "paddingVertical": 14,
    }
  }
>
  <View
    accessibilityRole="checkbox"
    accessibilityState={
      Object {
        "checked": false,
        "disabled": false,
      }
    }
    accessibilityValue={
      Object {
        "text": "off",
      }
    }
    accessible={true}
    pointerEvents="auto"
    testID="completeEdit"
  >
    <RNCCheckbox
      forwardedRef={null}
      onValueChange={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "height": 32,
            "width": 32,
          },
          undefined,
        ]
      }
      value={false}
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderColor": "#888888",
        "borderRadius": 6,
        "borderWidth": 1,
        "flex": 0.82,
        "justifyContent": "center",
        "marginHorizontal": 4,
        "position": "relative",
      }
    }
  >
    <TextInput
      keyboardType="default"
      onBlur={[Function]}
      onChangeText={[Function]}
      onEndEditing={[Function]}
      placeholder="Enter new task"
      style={
        Array [
          Object {
            "color": "#888888",
            "fontWeight": "600",
            "maxHeight": 120,
            "minHeight": 40,
            "padding": 10,
          },
        ]
      }
      testID="input"
      value=""
    />
    <Text
      style={
        Object {
          "color": "#ff2200",
          "fontSize": 12,
          "left": 45,
          "position": "absolute",
          "top": 44,
        }
      }
    >
      Name of task is mandatory!
    </Text>
  </View>
  <View
    accessibilityState={
      Object {
        "disabled": true,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#c8c8c8",
        "borderRadius": 6,
        "flex": 0.5,
        "justifyContent": "center",
        "marginHorizontal": 4,
        "maxHeight": 68,
        "maxWidth": 184,
        "minHeight": 40,
        "opacity": 1,
        "padding": 10,
      }
    }
    testID="createButton"
  >
    <Text
      style={
        Object {
          "color": "#fff",
          "fontWeight": "600",
          "textAlign": "center",
        }
      }
    >
      Edit Task
    </Text>
  </View>
</View>
`;

exports[`AddNewItem edit mode tests should handle user input 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "row",
      "justifyContent": "center",
      "marginVertical": 4,
      "paddingHorizontal": 10,
      "paddingVertical": 14,
    }
  }
>
  <View
    accessibilityRole="checkbox"
    accessibilityState={
      Object {
        "checked": false,
        "disabled": false,
      }
    }
    accessibilityValue={
      Object {
        "text": "off",
      }
    }
    accessible={true}
    pointerEvents="auto"
    testID="completeEdit"
  >
    <RNCCheckbox
      forwardedRef={null}
      onValueChange={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "height": 32,
            "width": 32,
          },
          undefined,
        ]
      }
      value={false}
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderColor": "#888888",
        "borderRadius": 6,
        "borderWidth": 1,
        "flex": 0.82,
        "justifyContent": "center",
        "marginHorizontal": 4,
        "position": "relative",
      }
    }
  >
    <TextInput
      keyboardType="default"
      onBlur={[Function]}
      onChangeText={[Function]}
      onEndEditing={[Function]}
      placeholder="Enter new task"
      style={
        Array [
          Object {
            "color": "#888888",
            "fontWeight": "600",
            "maxHeight": 120,
            "minHeight": 40,
            "padding": 10,
          },
        ]
      }
      testID="input"
      value="sampleTask4"
    />
  </View>
  <View
    accessibilityState={
      Object {
        "disabled": false,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#1c90ff",
        "borderRadius": 6,
        "flex": 0.5,
        "justifyContent": "center",
        "marginHorizontal": 4,
        "maxHeight": 68,
        "maxWidth": 184,
        "minHeight": 40,
        "opacity": 1,
        "padding": 10,
      }
    }
    testID="createButton"
  >
    <Text
      style={
        Object {
          "color": "#fff",
          "fontWeight": "600",
          "textAlign": "center",
        }
      }
    >
      Edit Task
    </Text>
  </View>
</View>
`;

exports[`AddNewItem edit mode tests should render component 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "row",
      "justifyContent": "center",
      "marginVertical": 4,
      "paddingHorizontal": 10,
      "paddingVertical": 14,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderColor": "#888888",
        "borderRadius": 6,
        "borderWidth": 1,
        "flex": 0.82,
        "justifyContent": "center",
        "marginHorizontal": 4,
        "position": "relative",
      }
    }
  >
    <TextInput
      keyboardType="default"
      onBlur={[Function]}
      onChangeText={[Function]}
      onEndEditing={[Function]}
      placeholder="Enter new task"
      style={
        Array [
          Object {
            "color": "#888888",
            "fontWeight": "600",
            "maxHeight": 120,
            "minHeight": 40,
            "padding": 10,
          },
        ]
      }
      testID="input"
      value=""
    />
  </View>
  <View
    accessibilityState={
      Object {
        "disabled": false,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#1c90ff",
        "borderRadius": 6,
        "flex": 0.5,
        "justifyContent": "center",
        "marginHorizontal": 4,
        "maxHeight": 68,
        "maxWidth": 184,
        "minHeight": 40,
        "opacity": 1,
        "padding": 10,
      }
    }
    testID="createButton"
  >
    <Text
      style={
        Object {
          "color": "#fff",
          "fontWeight": "600",
          "textAlign": "center",
        }
      }
    >
      Add Task
    </Text>
  </View>
</View>
`;

exports[`AddNewItem tests should handle add new after correct user input 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "row",
      "justifyContent": "center",
      "marginVertical": 4,
      "paddingHorizontal": 10,
      "paddingVertical": 14,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderColor": "#888888",
        "borderRadius": 6,
        "borderWidth": 1,
        "flex": 0.82,
        "justifyContent": "center",
        "marginHorizontal": 4,
        "position": "relative",
      }
    }
  >
    <TextInput
      keyboardType="default"
      onBlur={[Function]}
      onChangeText={[Function]}
      onEndEditing={[Function]}
      placeholder="Enter new task"
      style={
        Array [
          Object {
            "color": "#888888",
            "fontWeight": "600",
            "maxHeight": 120,
            "minHeight": 40,
            "padding": 10,
          },
        ]
      }
      testID="input"
      value="sample task 1"
    />
  </View>
  <View
    accessibilityState={
      Object {
        "disabled": false,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#1c90ff",
        "borderRadius": 6,
        "flex": 0.5,
        "justifyContent": "center",
        "marginHorizontal": 4,
        "maxHeight": 68,
        "maxWidth": 184,
        "minHeight": 40,
        "opacity": 1,
        "padding": 10,
      }
    }
    testID="createButton"
  >
    <Text
      style={
        Object {
          "color": "#fff",
          "fontWeight": "600",
          "textAlign": "center",
        }
      }
    >
      Add Task
    </Text>
  </View>
</View>
`;

exports[`AddNewItem tests should handle check for errors after incorrect user input 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "row",
      "justifyContent": "center",
      "marginVertical": 4,
      "paddingHorizontal": 10,
      "paddingVertical": 14,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderColor": "#888888",
        "borderRadius": 6,
        "borderWidth": 1,
        "flex": 0.82,
        "justifyContent": "center",
        "marginHorizontal": 4,
        "position": "relative",
      }
    }
  >
    <TextInput
      keyboardType="default"
      onBlur={[Function]}
      onChangeText={[Function]}
      onEndEditing={[Function]}
      placeholder="Enter new task"
      style={
        Array [
          Object {
            "color": "#888888",
            "fontWeight": "600",
            "maxHeight": 120,
            "minHeight": 40,
            "padding": 10,
          },
        ]
      }
      testID="input"
      value=""
    />
    <Text
      style={
        Object {
          "color": "#ff2200",
          "fontSize": 12,
          "left": 45,
          "position": "absolute",
          "top": 44,
        }
      }
    >
      Name of task is mandatory!
    </Text>
  </View>
  <View
    accessibilityState={
      Object {
        "disabled": true,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#c8c8c8",
        "borderRadius": 6,
        "flex": 0.5,
        "justifyContent": "center",
        "marginHorizontal": 4,
        "maxHeight": 68,
        "maxWidth": 184,
        "minHeight": 40,
        "opacity": 1,
        "padding": 10,
      }
    }
    testID="createButton"
  >
    <Text
      style={
        Object {
          "color": "#fff",
          "fontWeight": "600",
          "textAlign": "center",
        }
      }
    >
      Add Task
    </Text>
  </View>
</View>
`;

exports[`AddNewItem tests should handle check for errors after incorrect user input and then correct is given 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "row",
      "justifyContent": "center",
      "marginVertical": 4,
      "paddingHorizontal": 10,
      "paddingVertical": 14,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderColor": "#888888",
        "borderRadius": 6,
        "borderWidth": 1,
        "flex": 0.82,
        "justifyContent": "center",
        "marginHorizontal": 4,
        "position": "relative",
      }
    }
  >
    <TextInput
      keyboardType="default"
      onBlur={[Function]}
      onChangeText={[Function]}
      onEndEditing={[Function]}
      placeholder="Enter new task"
      style={
        Array [
          Object {
            "color": "#888888",
            "fontWeight": "600",
            "maxHeight": 120,
            "minHeight": 40,
            "padding": 10,
          },
        ]
      }
      testID="input"
      value="samplee task 2"
    />
  </View>
  <View
    accessibilityState={
      Object {
        "disabled": false,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#1c90ff",
        "borderRadius": 6,
        "flex": 0.5,
        "justifyContent": "center",
        "marginHorizontal": 4,
        "maxHeight": 68,
        "maxWidth": 184,
        "minHeight": 40,
        "opacity": 1,
        "padding": 10,
      }
    }
    testID="createButton"
  >
    <Text
      style={
        Object {
          "color": "#fff",
          "fontWeight": "600",
          "textAlign": "center",
        }
      }
    >
      Add Task
    </Text>
  </View>
</View>
`;

exports[`AddNewItem tests should handle error after incorrect user input 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "row",
      "justifyContent": "center",
      "marginVertical": 4,
      "paddingHorizontal": 10,
      "paddingVertical": 14,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderColor": "#888888",
        "borderRadius": 6,
        "borderWidth": 1,
        "flex": 0.82,
        "justifyContent": "center",
        "marginHorizontal": 4,
        "position": "relative",
      }
    }
  >
    <TextInput
      keyboardType="default"
      onBlur={[Function]}
      onChangeText={[Function]}
      onEndEditing={[Function]}
      placeholder="Enter new task"
      style={
        Array [
          Object {
            "color": "#888888",
            "fontWeight": "600",
            "maxHeight": 120,
            "minHeight": 40,
            "padding": 10,
          },
        ]
      }
      testID="input"
      value=""
    />
    <Text
      style={
        Object {
          "color": "#ff2200",
          "fontSize": 12,
          "left": 45,
          "position": "absolute",
          "top": 44,
        }
      }
    >
      Name of task is mandatory!
    </Text>
  </View>
  <View
    accessibilityState={
      Object {
        "disabled": true,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#c8c8c8",
        "borderRadius": 6,
        "flex": 0.5,
        "justifyContent": "center",
        "marginHorizontal": 4,
        "maxHeight": 68,
        "maxWidth": 184,
        "minHeight": 40,
        "opacity": 1,
        "padding": 10,
      }
    }
    testID="createButton"
  >
    <Text
      style={
        Object {
          "color": "#fff",
          "fontWeight": "600",
          "textAlign": "center",
        }
      }
    >
      Add Task
    </Text>
  </View>
</View>
`;

exports[`AddNewItem tests should handle user input 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "row",
      "justifyContent": "center",
      "marginVertical": 4,
      "paddingHorizontal": 10,
      "paddingVertical": 14,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderColor": "#888888",
        "borderRadius": 6,
        "borderWidth": 1,
        "flex": 0.82,
        "justifyContent": "center",
        "marginHorizontal": 4,
        "position": "relative",
      }
    }
  >
    <TextInput
      keyboardType="default"
      onBlur={[Function]}
      onChangeText={[Function]}
      onEndEditing={[Function]}
      placeholder="Enter new task"
      style={
        Array [
          Object {
            "color": "#888888",
            "fontWeight": "600",
            "maxHeight": 120,
            "minHeight": 40,
            "padding": 10,
          },
        ]
      }
      testID="input"
      value="sampleTask4"
    />
  </View>
  <View
    accessibilityState={
      Object {
        "disabled": false,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#1c90ff",
        "borderRadius": 6,
        "flex": 0.5,
        "justifyContent": "center",
        "marginHorizontal": 4,
        "maxHeight": 68,
        "maxWidth": 184,
        "minHeight": 40,
        "opacity": 1,
        "padding": 10,
      }
    }
    testID="createButton"
  >
    <Text
      style={
        Object {
          "color": "#fff",
          "fontWeight": "600",
          "textAlign": "center",
        }
      }
    >
      Add Task
    </Text>
  </View>
</View>
`;

exports[`AddNewItem tests should remove error due to incorrect user input, after correct one is given 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "row",
      "justifyContent": "center",
      "marginVertical": 4,
      "paddingHorizontal": 10,
      "paddingVertical": 14,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderColor": "#888888",
        "borderRadius": 6,
        "borderWidth": 1,
        "flex": 0.82,
        "justifyContent": "center",
        "marginHorizontal": 4,
        "position": "relative",
      }
    }
  >
    <TextInput
      keyboardType="default"
      onBlur={[Function]}
      onChangeText={[Function]}
      onEndEditing={[Function]}
      placeholder="Enter new task"
      style={
        Array [
          Object {
            "color": "#888888",
            "fontWeight": "600",
            "maxHeight": 120,
            "minHeight": 40,
            "padding": 10,
          },
        ]
      }
      testID="input"
      value="sampletask22"
    />
    <Text
      style={
        Object {
          "color": "#ff2200",
          "fontSize": 12,
          "left": 45,
          "position": "absolute",
          "top": 44,
        }
      }
    >
      Name of task is mandatory!
    </Text>
  </View>
  <View
    accessibilityState={
      Object {
        "disabled": true,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#c8c8c8",
        "borderRadius": 6,
        "flex": 0.5,
        "justifyContent": "center",
        "marginHorizontal": 4,
        "maxHeight": 68,
        "maxWidth": 184,
        "minHeight": 40,
        "opacity": 1,
        "padding": 10,
      }
    }
    testID="createButton"
  >
    <Text
      style={
        Object {
          "color": "#fff",
          "fontWeight": "600",
          "textAlign": "center",
        }
      }
    >
      Add Task
    </Text>
  </View>
</View>
`;

exports[`AddNewItem tests should render component 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "row",
      "justifyContent": "center",
      "marginVertical": 4,
      "paddingHorizontal": 10,
      "paddingVertical": 14,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderColor": "#888888",
        "borderRadius": 6,
        "borderWidth": 1,
        "flex": 0.82,
        "justifyContent": "center",
        "marginHorizontal": 4,
        "position": "relative",
      }
    }
  >
    <TextInput
      keyboardType="default"
      onBlur={[Function]}
      onChangeText={[Function]}
      onEndEditing={[Function]}
      placeholder="Enter new task"
      style={
        Array [
          Object {
            "color": "#888888",
            "fontWeight": "600",
            "maxHeight": 120,
            "minHeight": 40,
            "padding": 10,
          },
        ]
      }
      testID="input"
      value=""
    />
  </View>
  <View
    accessibilityState={
      Object {
        "disabled": false,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#1c90ff",
        "borderRadius": 6,
        "flex": 0.5,
        "justifyContent": "center",
        "marginHorizontal": 4,
        "maxHeight": 68,
        "maxWidth": 184,
        "minHeight": 40,
        "opacity": 1,
        "padding": 10,
      }
    }
    testID="createButton"
  >
    <Text
      style={
        Object {
          "color": "#fff",
          "fontWeight": "600",
          "textAlign": "center",
        }
      }
    >
      Add Task
    </Text>
  </View>
</View>
`;
